frontend/src/components/Dashboard/ScoresTableSection/index.jsx

```diff
@@ -2,78 +2,86 @@ import React from 'react'
 import { connect } from 'react-redux'
 import { Table, TableBody, TableHeader, TableHeaderColumn, TableRow, TableRowColumn, TextField } from 'material-ui'
 
-import { fetchSubjectsList, fetchStudentsList, fetchScoresList } from 'actions/group'
+import { fetchSubjectsList, fetchStudentsList, fetchScoresList, applyNewScore } from 'actions/group'
 
 
 class ScoresTable extends React.Component {
   constructor(props) {
     super(props)
     this.state = {
-      subjectsList: null,
-      studentsList: null,
-      scoresList: null,
-      neededSubjects: null,
-      rebuiltStudentsList: null,
+      groupSubjectsList: null,
+      rebuiltSubjectsList: null,
     }
-    this.handleInput = this.handleInput.bind(this)
   }
 
-  handleInput(event){
+  handleInput(studentId, courseId, event){
+    const { applyNewScore } = this.props
     const inputValue = event.target.value
```

Comment:
As the app grows, it'll need more memory to store all of the refs. Whereas the event solution doesn't need to store any data (except of the current cell entered data) at all.